docs/source/tutorial/tutorial-detail-view.md

```diff
@@ -0,0 +1,242 @@
+  let rocketType = launch.rocket?.type {
+    self.rocketNameLabel.text = "🚀 \(rocketName) (\(rocketType))"
+} else {
+  self.rocketNameLabel.text = nil
+}
+    
+if launch.isBooked {
+  self.bookCancelButton.title = "Cancel trip"
+  self.bookCancelButton.tintColor = .red
+} else {
+  self.bookCancelButton.title = "Book now!"
+  // Get the color from the main window rather than the view to prevent alerts from draining color
+  self.bookCancelButton.tintColor = UIApplication.shared.windows.first?.tintColor
+}
+```
+
+Then, find the `loadLaunchDetails()`  method. Replace the `TODO` with the following, which loads the details using the `LaunchDetailsQuery` you created earlier:
+
+```swift:title=DetailViewController.swift
+private func loadLaunchDetails() {
+  guard
+    let launchID = self.launchID,
+    launchID != self.launch?.id else {
+      // This is the launch we're already displaying, or the ID is nil.
+      return
+  }
+    
+  Network.shared.apollo.fetch(query: LaunchDetailsQuery(launchId: launchID)) { [weak self] result in
+    guard let self = self else {
+      return
+    }
+    
+    switch result {
+    case .failure(let error):
+      self.showAlert(title: "Network Error",
+                     message: error.localizedDescription)
+    case .success(let graphQLResult):
+      if let launch = graphQLResult.data?.launch {
+        self.launch = launch
+      }
+    
+      if let errors = graphQLResult.errors {
+        let message = errors
+                        .map { $0.localizedDescription }
+                        .joined(separator: "\n")
+        self.showAlert(title: "GraphQL Error(s)",
+                       message: message)
+      }
+    }
+  }
+}
+```
+
+Finally, update the `didSet` for `launchID` to load the launch details if we don't already have them: 
+
+```swift:title=DetailViewController.swift
+var launchID: GraphQLID? {
+  didSet {
+    self.loadLaunchDetails()
+  }
+}
+```
+
+and add a `didSet` on the `launch` property to load the UI once the launch is actually loaded. 
+
+```swift:title=DetailViewController.swift
+private var launch: LaunchDetailsQuery.Data.Launch? {
+  didSet {
+    self.configureView()
+  }
+}
+```
+
+Build and run the application. When you tap into the detail screen, you should now see the full details: 
+
+<img src="images/detail_loaded.png" alt="Final display of the detail screen" class="screenshot" width="300"/>
+
+You'll notice that many of the more recent launches have a rocket type of `FT`. If you load more launches until you get to the end of the list, you'll get to some rockets that have different rocket types: 
+
+<img src="images/detail_loaded_merlin_c.png" alt="Final display with different rocket type" class="screenshot" width="300"/>
+
+You may have noticed that the detail view includes a `Book Now!` button, but there's no way to book a seat yet. To fix that, let's [learn how to make changes to objects in your graph with mutations](./tutorial-mutations).
```

Comment:
This is Step 6 of the tutorial but this "mutations" link jumps to Step 8; the "Next" link below this content still correctly goes to Step 7.

Reply:
Great catch, leftover from before that got broken out.